src/component/Canvas.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import Node from './Node';
+import Edge from './Edge';
+
+function calulateCoordinates(
+  nodes: string[],
+  edges: { from: number, to: number }[]) {
+  // super complicated algorithm calculating optimal coordinates
+  const mappedNodes = nodes.map((l, i) => ({
+    label: l,
+    id: i,
+    x: Math.random() * 100,
+    y: Math.random() * 100,
+  }));
+  const mappedEdges = edges.map((obj, i) => {
+    const node1 = mappedNodes.find((e) => e.id === obj?.from);
+    const node2 = mappedNodes.find((e) => e.id === obj?.to);
+    return {
+      id: i,
+      x1: node1?.x ?? 0,
+      y1: node1?.y ?? 0,
+      x2: node2?.x ?? 0,
+      y2: node2?.y ?? 0,
+    };
+  });
+  console.log('mappedNodes', mappedNodes);
+  return {n: mappedNodes, e: mappedEdges};
+}
+
+function calculateEdgePosition(e: {
```

Comment:
I'd define an Interface for e and put int above, also please name it `edge` instead of `e` for good practice sake